ebms-provider/src/main/kotlin/no/nav/emottak/ebms/App.kt

```diff
@@ -45,24 +50,47 @@ fun main() = SuspendApp {
     val database = Database(ebmsDbConfig.value)
     database.migrate(ebmsMigrationConfig.value)
 
+    val config = config()
     val ebmsMessageDetailsRepository = EbmsMessageDetailsRepository(database)
+    val eventsRepository = EventsRepository(database)
     val payloadRepository = PayloadRepository(database)
+    val processingClient = PayloadProcessingClient(scopedAuthHttpClient(EBMS_PAYLOAD_SCOPE))
+    val processingService = ProcessingService(processingClient)
+    val ebmsSignalProducer = EbmsSignalProducer(config.kafkaSignalProducer.topic, config.kafka)
+
+    val cpaClient = CpaRepoClient(defaultHttpClient())
+    val dokumentValidator = DokumentValidator(cpaClient)
+
+    val smtpTransportClient = SmtpTransportClient(scopedAuthHttpClient(SMTP_TRANSPORT_SCOPE))
+
+    launchSignalReceiver(
+        config,
+        ebmsMessageDetailsRepository,
+        dokumentValidator
+    )
+    launchPayloadReceiver(
+        config,
+        ebmsMessageDetailsRepository,
+        eventsRepository,
+        dokumentValidator,
+        processingService,
+        ebmsSignalProducer,
+        smtpTransportClient
+    )
```

Comment:
Jeg har en tanke og et spørsmål: Ville det være mulig å lage en separat konfigurasjonsmodul for dette? Når jeg leser gjennom koden, virker det som vi introduserer mange variabler og funksjoner for ett eller to formål. Jeg tror det kunne vært fordelaktig å rydde opp i App.kt ved å separere ut relatert kode inn i egne filer og klasser. 
Dette kan forbedre lesbarheten og vedlikeholdbarheten. Hva tenker du?

Reply:
Lurer litt på om sync og async skal splittes opp mye mer enn det som er gjort her. Kanskje egne deployments, slik at async-flyten ikke påvirker sync-flyten og motsatt. Men har valgt å ikke forskuttere noen avgjørelser, og holder derfor konfigurasjonen forholdsvis enkel foreløpig.